File: backend/voting-app/src/main/java/com/tu/votingapp/dto/general/RoleDTO.java
package com.tu.votingapp.dto.general;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RoleDTO {
    private Long id;
    private String name;
    private List<PermissionDTO> permissions;
}

File: frontend/android/app/src/main/java/com/example/android/dto/response/RoleDTO.kt
package com.example.android.dto.response

import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

@Serializable
data class RoleDTO(
    val id: Long? = null,
    val name: String,
    @Transient val permissions: List<PermissionDTO> = emptyList()
)